Description: skip disk/by-uuid for bcache devices
 blkid reports bcache superblock dev.uuid as a filesystem UUID but it actually
 is not a filesystem, it's the UUID of the backing device, which is maintained
 at /dev/bcache/by-uuid instead of /dev/disk/by-uuid.
Author: Ryan Harper <ryan.harper@canonical.com>
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/bcache-tools/+bug/1861941
Forwarded: no
Last-Update: 2020-05-22
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/rules.d/60-persistent-storage.rules
+++ b/rules.d/60-persistent-storage.rules
@@ -109,8 +109,11 @@ KERNEL=="sr*", ENV{DISK_EJECT_REQUEST}!=
 KERNEL!="sr*", IMPORT{builtin}="blkid"
 
 # by-label/by-uuid links (filesystem metadata)
+# Skip bcache backing devices, handled in 69-bcache.rules
+ENV{ID_FS_TYPE}=="bcache", GOTO="skip_bcache_fs_type"
 ENV{ID_FS_USAGE}=="filesystem|other|crypto", ENV{ID_FS_UUID_ENC}=="?*", SYMLINK+="disk/by-uuid/$env{ID_FS_UUID_ENC}"
 ENV{ID_FS_USAGE}=="filesystem|other|crypto", ENV{ID_FS_LABEL_ENC}=="?*", SYMLINK+="disk/by-label/$env{ID_FS_LABEL_ENC}"
+LABEL="skip_bcache_fs_type"
 
 # by-id (World Wide Name)
 ENV{DEVTYPE}=="disk", ENV{ID_WWN_WITH_EXTENSION}=="?*", SYMLINK+="disk/by-id/wwn-$env{ID_WWN_WITH_EXTENSION}"
